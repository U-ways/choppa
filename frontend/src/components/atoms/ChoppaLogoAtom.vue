<template>
<font-awesome icon="helicopter" :class="[ this.css ]" />
</template>

<script setup="props" lang="ts">
declare const props: { //eslint-disable-line no-unused-vars
  css: string
};
</script>

<style scoped lang="scss">
</style>