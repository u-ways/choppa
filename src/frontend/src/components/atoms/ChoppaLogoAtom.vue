<template>
  <div>
    <font-awesome-icon icon="helicopter" :class="[ this.css ]" />
  </div>
</template>

<script>
export default {
  name: "ChoppaLogoAtom",
  props: {
    css: String,
  },
};
</script>

<style scoped>
</style>
