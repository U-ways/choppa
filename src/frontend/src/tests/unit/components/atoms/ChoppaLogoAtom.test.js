import { describe, expect, it, shallowMount } from "@/tests/unit/test-imports";
import ChoppaLogoAtom from "@/components/atoms/ChoppaLogoAtom";

describe("Choppa Logo Atom test", () => {
  it("Renders correctly", async () => {
    const wrapper = shallowMount(ChoppaLogoAtom, {
      propsData: {
        css: "my-css-test",
      },
      stubs: {
        "font-awesome-icon": true,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
