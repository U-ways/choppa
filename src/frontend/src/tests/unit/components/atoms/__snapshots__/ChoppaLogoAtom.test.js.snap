// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choppa Logo Atom test Renders correctly 1`] = `
"<div>
  <font-awesome-icon-stub icon=\\"helicopter\\" class=\\"my-css-test\\"></font-awesome-icon-stub>
</div>"
`;
